projects/js-packages/publicize-components/src/hooks/use-anaytics-tracks/index.js

```diff
@@ -0,0 +1,90 @@
+import jetpackAnalytics from '@automattic/jetpack-analytics';
+import { useConnection } from '@automattic/jetpack-connection';
+import { useCallback, useEffect } from 'react';
+
+const useAnalyticsTracks = ( {
+	pageViewEventName,
+	pageViewNamespace = 'jetpack',
+	pageViewSuffix = 'page_view',
+	pageViewEventProperties = {},
+} = {} ) => {
+	const { isUserConnected, isRegistered, userConnectionData } = useConnection();
+	const { login, ID } = userConnectionData.currentUser?.wpcomUser || {};
+	const { tracks } = jetpackAnalytics;
+	const { recordEvent } = tracks;
```

Comment:
These two consts can be defined outside of the scope of the hook function - reducing recalculations happening on every rerender.

Reply:
Nice improvement, thanks for spotting! 👍